### ISSUE TITLE
[Info] Air purifier 2S aka zhimi.airpurifier.mc1

### ISSUE THREAD

**Issue by Sineos** [JoJ123/ioBroker.mihome-airpurifier#11]
Unfortunately the miio library currently does not support the 2S model with the device string `zhimi.airpurifier.mc1` and the newer firmwares do not allow to get the hex token with the `miio discover` command.

To make the adapter work with this device, follow these steps:
1. Install the adapter
2. Login via SSH to your ioBroker server
3. `cd /opt/iobroker/node_modules/iobroker.mihome-airpurifier/node_modules/miio/lib`
4. `nano models.js`
5. Insert `'zhimi.airpurifier.mc1': AirPurifier,` (mind the coma!) just below the `// Air Purifier 2S` section
6. Upload the modified adapter with the command `iobroker upload mihome-airpurifier`
7. Follow [this instructions](https://mensfeld.pl/2018/11/extracting-the-device-token-from-xiaomi-air-purifier-2s-eu-for-domoticz-usage/) to get the hex token (Android only). With IOS you can achieve the same with an unencrypted iTunes backup

**Comment by Sineos** [comment]
Additional information:

This [Russian site](http://www.kapiba.ru/2017/11/mi-home.html) has a modded Android Mi Home apk. This shows the `token` under _Network Information_ of the respective devices.

Warning:
A quick Virus Total scan did not show any malicious results. Nevertheless be careful with unknown/modded apps.

**Comment by JoJ123** [comment]
There's already an open PR for the miio library.
https://github.com/aholstenson/miio/pull/185

Should be fixed soon in this adapter

**Comment by Sineos** [comment]
I wish I would share your optimistic view. For me this library unfortunately seems pretty unmaintained. The PR is open since October and the last committ is May.

**Comment by Sineos** [comment]
https://github.com/jghaanstra/miio/commits/master would be an option as it seems active.

**Comment by JoJ123** [comment]
@Sineos Can you please try this branch in your ioBroker installation:
https://github.com/JoJ123/ioBroker.mihome-airpurifier/tree/fix/miioVersion

I've included the miio from jghaanstra in the package.json

**Comment by Sineos** [comment]
Basically it works, except that a recent commit added a small bug leading to a crash. See https://github.com/jghaanstra/miio/pull/2

Controls also work for me, except the `level` control. New value is not ack'd and the device does not change

```
mihome-airpurifier.0 | 2019-04-08 19:01:48.451 | debug | mode: favorite
mihome-airpurifier.0 | 2019-04-08 19:01:48.222 | debug | stateChange mihome-airpurifier.0.control.manuallevel  {"val":100,"ack":false,"ts":1554750108220,"q":0,"from":"system.adapter.admin.0","user":"system.user.admin","lc":1554750108220}
```

**Comment by Sineos** [comment]
Following the discussion in the a.m. PR, the @jghaanstra fork is also a bad choice. 

My vote would be forking this fork, correcting the modified humidifier2.js (irrelevant for the purifier series anyway) and using this one. 
Should anyone pick-up the development or bring good addition, they could be cherry picked or the dependency changed.

The above mentioned issue about setting the level was false alarm: Works in current master as well as in /fix/miioVersion

**Comment by JoJ123** [comment]
Can you create the fork? Then I will integrate your fork an we go with this
to master & npm :)

Sineos <notifications@github.com> schrieb am Di., 9. Apr. 2019, 20:52:

**Comment by Sineos** [comment]
Be my guest :-)
https://github.com/Sineos/miio

**Comment by JoJ123** [comment]
@Sineos  I've update the branch fix/miioVersion
Can you please give it a try again, then I will merge it to master.

**Comment by Sineos** [comment]
Works nicely on my side 👍

**Comment by JoJ123** [comment]
merged and available on npm in version 0.0.6 :) Thx for the help.
